gnome-image-installer/util/gis-write-diagnostics.c

```diff
@@ -0,0 +1,205 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gis-write-diagnostics.h"
+
+typedef struct {
+    gchar *eos_diagnostics_exe;
+    GFile *image_dir;
+    gchar *home_dir;
+} WriteDiagnosticsData;
+
+static WriteDiagnosticsData *
+write_diagnostics_data_new (const gchar *eos_diagnostics_exe,
+                            GFile       *image_dir,
+                            const gchar *home_dir)
+{
+  WriteDiagnosticsData *d = g_new (WriteDiagnosticsData, 1);
+  d->eos_diagnostics_exe = g_strdup (eos_diagnostics_exe ?: "eos-diagnostics");
+  d->image_dir = image_dir != NULL ? g_object_ref (image_dir) : NULL;
+  d->home_dir = g_strdup (home_dir ?: NULL);
```

Comment:
The ternary seems a bit pointless here, since `home_dir` is either set or `NULL`.

Reply:
Unless it’s there to make it more obvious that `home_dir` _could_ be `NULL`, in which case: fine, carry on.

Reply:
This was leftover from another incarnation of this function I think.